### PR TITLE
Feature: DAO 613 - Progress status

### PR DESCRIPTION
**Pull request by WissenIstNacht**
This PR implements:
- Progress->Status component
- Corresponding story
- Corresponding test

### CODE REVIEW

File: packages/ui-components/src/components/icons/interface/icon_block.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      width={width}
+      height={height}
+      fill="none"
+      viewBox="0 0 16 16"
+      {...props}
+    >
+      <g clip-path="url(#clip0_433_16)">
+        <path
+          fill-rule="evenodd"
+          clip-rule="evenodd"
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/icons/interface/icon_block.tsx]: Can you camel case `fill-rule` and `clip-rule`?

File: packages/ui-components/src/components/icons/interface/radio/cancle.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}) => {
+  return (
+    <svg
+      width={width}
+      height={height}
+      viewBox="0 0 16 16"
+      fill="none"
+      {...props}
+    >
+      <g clip-path="url(#clip0_430_5)">
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/icons/interface/radio/cancle.tsx]: Camel casing for `clip-path`

File: packages/ui-components/src/components/icons/interface/radio/cancle.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import {IconType} from '../..';
+
+export const IconRadioCancle: IconType = ({
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/icons/interface/radio/cancle.tsx]: Slight misspelling, should be 'cancel'.

File: packages/ui-components/src/components/progress/status.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    <TopContainer data-testid="progressStatus">
+      <LeftContainer mode={mode}>
+        <IconContainer>
+          <Icon mode={mode} />
+        </IconContainer>
+        <LabelContainer>
+          <CustomLabel label={label} helpText={date} />
+        </LabelContainer>
+      </LeftContainer>
+      <BlockContainer>
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/progress/status.tsx]: If no block is given, the icon shouldn't be rendered.

File: packages/ui-components/src/components/progress/status.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    case 'failed':
+      return <IconRadioCancle className={iconColors[mode]} />;
+    default:
+      return <IconSuccess className={iconColors[mode]} />;
+  }
+};
+
+type CustomLabelProps = Pick<LabelProps, 'label' | 'helpText'>;
+
+/* Lord knows it hurts. */
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/progress/status.tsx:129]: Yes, this is just sad, really.

File: packages/ui-components/src/components/progress/status.tsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+   * it is in the future).
+   *
+   * If no date is passed when one is required, a fallback text will be displayed.
+   * */
+  date?: string;
+  /**
+   * If the progress status changed due to an event on a blockchain, the
+   * corresponding block MAY be passed. Note that upcoming, active and rejected
+   * mode can NEVER have a block associated.
+   */
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/progress/status.tsx:44]: I appreciate this comment. Is this something that can or even should be implemented in the code itself? Like if the mode is one of that shouldn't show the block, hide the block even if the property is provided? While it does limit the user, it also doesn't ever make sense to have the block shown. Thoughts?

File: packages/ui-components/src/components/progress/status.tsx [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+   * Describes the name of the progress step. Think of it as the name of the
+   * todo in the example above.
+   */
+  label: string;
+  /**
+   * Describes when the progress status was last changed. Every mode of progress
+   * status MUST have a date, EXCEPT for "upcoming", which NEVER has a date, (as
+   * it is in the future).
+   *
+   * If no date is passed when one is required, a fallback text will be displayed.
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/progress/status.tsx:37]: I think you either forgot to add the fallback text or update the comment!

Also, would hiding the date even if it is given when the mode is `upcoming` a better approach? Related to the block prop comment.

File: packages/ui-components/src/components/progress/status.tsx [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+  IconBlock,
+  IconRadioCancle,
+  IconRadioDefault,
+  IconSpinner,
+  IconSuccess,
+} from '../icons';
+import {LabelProps} from '../label';
+
+type ModeType = 'active' | 'failed' | 'done' | 'succeeded' | 'upcoming';
+export type ProgressStatusProps = {
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/progress/status.tsx:13]: Very thorough documentation; I like! 👍🏾

File: packages/ui-components/src/components/progress/status.tsx [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+      </BlockContainer>
+    </TopContainer>
+  );
+};
+
+type ModeProps = {
+  mode: ModeType;
+};
+
+const TopContainer = styled.div.attrs({className: 'flex justify-between'})``;
```
Comment (Fabricevladimir) [inline packages/ui-components/src/components/progress/status.tsx:83]: This seems to be a fixed width on the designs. Currently, the content is squished together on mobile.

**Review by Fabricevladimir — CHANGES_REQUESTED** [review]
Nice work, thanks! Left a few comments here and there. I'm pleasantly surprised at how complicated the component itself turned out to be with the different props affecting each other.

**Review by RakeshUP — APPROVED** [review]
Nice. The whole BlockContainer should be conditionally rendered. The other comments are very minor too.

File: packages/ui-components/src/components/progress/status.tsx (hunk 9 above)
Comment (WissenIstNacht) [reply packages/ui-components/src/components/progress/status.tsx:83]: Hmm. You're right. But does that make sense...?

FYI: The reason for the squishing on mobile is not in the component itself but in the way I set up the story. I wrapped it into a div that takes 1/3 of the screen. On mobile, that's obviously way too small and squishes the content. I've modified the story to only do the 1/3 width on tablet or larger.

Should I still fix the width or do you think it makes sense after the correction?

File: packages/ui-components/src/components/progress/status.tsx (hunk 7 above)
Comment (WissenIstNacht) [reply packages/ui-components/src/components/progress/status.tsx:37]: Turns out I pretty much forgot to do all of that stuff. Thanks for catching it!

File: packages/ui-components/src/components/progress/status.tsx (hunk 9 above)
Comment (Fabricevladimir) [reply packages/ui-components/src/components/progress/status.tsx:83]: Ah, that makes sense. I'm not sure why it's fixed width (maybe Figma thing?). To me it makes more sense having it take up the size of the parent.

**Review by Fabricevladimir — APPROVED** [review]

**Pull request merged**